### PR TITLE
Use array-api-extra cov() (PR #690) for batched covariance estimation

### PR DESCRIPTION
**Pull request by bruAristimunha**
## What

Route the `"cov"` covariance estimator through
[`array_api_extra.cov`](https://data-apis.org/array-api-extra/), using the
new parameters added in **[data-apis/array-api-extra#690](https://github.com/data-apis/array-api-extra/pull/690)**
(`axis` / `correction` / `fweights` / `aweights`).

This is a small, concrete downstream consumer of that PR — see the
[motivating comment](https://github.com/data-apis/array-api-extra/pull/690#issuecomment-5057787560).

Solve a part of https://github.com/pyRiemann/pyRiemann/issues/465

## Why

`pyriemann.geometry.covariance._cov` used to:

1. hand-translate `bias`/`ddof` → `correction` (the `_cov_kwargs_to_xp` /
   `apply_xp_cov` shim in `_backend.py`) to paper over the NumPy-vs-PyTorch
   keyword split, and
2. loop over every leading batch dimension in Python (`_vectorize_nd`).

`array_api_extra.cov` now does **both** upstream: it takes the array-API
standard `correction` keyword and estimates covariances for all leading
batch dimensions in a single call, across backends. So for the `"cov"`
path we can delegate and drop the local machinery:

```python
def _cov(X, **kwds):
    xp = get_namespace(X)
    cov = xpx_cov(X, axis=-1, **_cov_kwargs_to_xp(kwds))
    return xp.reshape(cov, (*X.shape[:-2], X.shape[-2], X.shape[-2]))
```

Observations are the time samples, hence `axis=-1`. The reshape restores
the two matrix axes that `cov()` collapses when `n_channels == 1`, so the
output stays `(..., n_channels, n_channels)` in every case.

`_corr` still uses `apply_xp_cov(xp.corrcoef, ...)` — array-api-extra has no
`corrcoef` — so the shim stays for that estimator.

## Behavior

Output is **unchanged** for existing usage (`n_channels >= 2`): `xpx.cov`'s
defaults (`axis=-1, correction=1`) match `numpy.cov`, verified bit-for-bit
against a per-matrix `np.cov` loop up to 5-D batches, on both NumPy and
PyTorch. The only new coverage is the `n_channels == 1` shape guard.

## Tests

- `tests/test_geometry_covariance.py`: **all green** on NumPy + PyTorch.
- Added `test_covariances_single_channel` locking the `(..., 1, 1)` shape.

## Dependency

Needs array-api-extra #690, not yet released. `setup.py` and
`requirements.txt` temporarily pin the branch:

```
array-api-extra @ git+https://github.com/bruAristimunha/array-api-extra@cov_parameters
```

Marked as **draft** — revert the pin to a released `>=` floor once #690 is
merged and published.

### CODE REVIEW

**Comment by bruAristimunha** [comment]
### Code review

Found 1 issue:

1. The default `"cov"` estimator silently breaks PyTorch autograd on batched input. Excluding `_cov` from `_vectorize_nd` feeds the full batched array into `array_api_extra.cov`, whose batched path runs `xp.asarray(m)`; `torch.asarray` defaults to `requires_grad=False`, detaching the graph. So `covariances(X)` with `X.requires_grad=True` and shape `(n_trials, n_channels, n_times)` returns `requires_grad=False`, and inside a larger graph `.backward()` runs but leaves `X.grad is None` (silent gradient loss). The 2-D path and `estimator="scm"` are unaffected. This is the autograd property that #433 added torch support to preserve, and no covariance autograd test currently guards it.

Root cause — the batched array now bypasses the per-slice `torch.cov` path:

https://github.com/pyRiemann/pyRiemann/blob/a6499c36a11cc5255c466889878729cd877b3ecf/pyriemann/geometry/covariance.py#L415-L418

Detaching call:

https://github.com/pyRiemann/pyRiemann/blob/a6499c36a11cc5255c466889878729cd877b3ecf/pyriemann/geometry/covariance.py#L29-L35

**Comment by bruAristimunha** [comment]
Update: the root cause is fixed upstream in `array-api-extra` (the `cov_parameters` branch this PR already pins), commit bruAristimunha/array-api-extra@b6e9e1c352d3b7dbeecc5983f01ad9f48d7a5bab.

The generic `cov` path called `xp.asarray(m)` unconditionally; on torch that detaches gradients and mutates the input tensor in place. It is now guarded to run only for genuine non-array input, so arrays pass through untouched. A torch autograd regression test was added to `array-api-extra`'s `TestCov`.

With that branch, batched `covariances(X)` with `X.requires_grad=True` returns `requires_grad=True` and `X.grad` flows (verified locally; values still match `numpy.cov`). So the native-batching change in this PR no longer regresses autograd.
